// qur/src/commands/base.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const readline = require("readline");
var rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});
// todo omelette
class Command {
    constructor() {
        this.state = {
            configuration: {
                exists() {
                    // todo 
                    return false;
                },
                create() {
                    // todo 
                }
            }
        };
        if (!this.state.configuration.exists()) {
            console.log('Just a second!', 'Seems that you are using Qur for the first time.', 'Do you prefer Typescript (recommended) or Javascript workflow?');
            rl.question('Write "ts" or "js" (ts)', (answer) => {
                console.log(answer); // TODO DELETE
            });
            // todo 
            console.log('conf file created '); // TODO DELETE
            this.state.configuration.create();
        }
    }
}
exports.default = Command;
